lib/orderbook/OrderBook.ts

```diff
@@ -404,17 +407,26 @@ class OrderBook extends EventEmitter {
       }
     }
 
-    if (this.pool.getNetwork() === XuNetwork.MainNet) {
-      // check if order abides by limits
-      const outboundCurrencyLimit = limits[outboundCurrency];
-      if (outboundCurrencyLimit && outboundAmount > outboundCurrencyLimit) {
-        throw errors.EXCEEDING_LIMIT(outboundCurrency, outboundAmount, outboundCurrencyLimit);
-      }
+    // check if order abides by limits
+    let outboundCurrencyLimit: number;
+    let inboundCurrencyLimit: number;
 
-      const inboundCurrencyLimit = limits[inboundCurrency];
-      if (inboundCurrencyLimit && inboundAmount > inboundCurrencyLimit) {
-        throw errors.EXCEEDING_LIMIT(inboundCurrency, inboundAmount, inboundCurrencyLimit);
-      }
+    if (this.pool.getNetwork() === XuNetwork.MainNet && !this.maxlimits) {
+      // if we're on mainnet and we haven't specified that we're using maximum limits
+      // then use the hardcoded mainnet order size limits
+      outboundCurrencyLimit = limits[outboundCurrency];
+      inboundCurrencyLimit = limits[inboundCurrency];
+    } else {
+      // otherwise use the maximum channel sizes as order size limits
+      outboundCurrencyLimit = maxLimits[outboundCurrency];
+      inboundCurrencyLimit = maxLimits[inboundCurrency];
+    }
+
+    if (outboundCurrencyLimit && outboundAmount > outboundCurrencyLimit) {
+      throw errors.EXCEEDING_LIMIT(outboundCurrency, outboundAmount, outboundCurrencyLimit);
+    }
+    if (inboundCurrencyLimit && inboundAmount > inboundCurrencyLimit) {
+      throw errors.EXCEEDING_LIMIT(inboundCurrency, inboundAmount, inboundCurrencyLimit);
```

Comment:
`errors.EXCEEDING_LIMIT` msg is misleading - it's referring the outbound amount, while here it's used for the inbound amount.

Reply:
Good call, I'll change the message to make it apply either way.